### PR TITLE
Remove storage session pre requisites

### PR DESCRIPTION
**Pull request by colin-higgins**
@Particular/tf-async-persistence 

Removed the prerequisites as we have discussed. Thoughts? Reviews? Pennies?

### CODE REVIEW

File: src/NServiceBus.RavenDB.AcceptanceTests/App_Packages/NSB.AcceptanceTests.6.0.0-unstable1448/Sagas/when_receiving_multiple_timeouts.cs [hunk 1]
```diff
@@ -38,6 +38,7 @@ public Endpoint()
             {
                 EndpointSetup<DefaultServer>(c =>
                 {
+                    c.EnableFeature<FirstLevelRetries>();
```
Comment (DavidBoike) [inline src/NServiceBus.RavenDB.AcceptanceTests/App_Packages/NSB.AcceptanceTests.6.0.0-unstable1448/Sagas/when_receiving_multiple_timeouts.cs]: Corresponding Core PR for this test modification?

File: src/NServiceBus.RavenDB/SessionManagement/RavenDbStorageSession.cs [hunk 2]
```diff
@@ -10,11 +10,6 @@
 
     class RavenDbStorageSession : Feature
     {
-        public RavenDbStorageSession()
-        {
-            DependsOnAtLeastOne(typeof(RavenDbSagaStorage), typeof(RavenDbOutboxStorage));
-        }
```
Comment (DavidBoike) [inline src/NServiceBus.RavenDB/SessionManagement/RavenDbStorageSession.cs]: Should there still be a constructor that enables by default? Or why is removing the the dependency information altogether the best strategy?

Comment (colin-higgins) [inline src/NServiceBus.RavenDB/SessionManagement/RavenDbStorageSession.cs]: Enables by default makes sense to me.

File: src/NServiceBus.RavenDB.AcceptanceTests/App_Packages/NSB.AcceptanceTests.6.0.0-unstable1448/Sagas/when_receiving_multiple_timeouts.cs (hunk 1 above)
Comment (colin-higgins) [inline src/NServiceBus.RavenDB.AcceptanceTests/App_Packages/NSB.AcceptanceTests.6.0.0-unstable1448/Sagas/when_receiving_multiple_timeouts.cs]: https://github.com/Particular/NServiceBus/pull/3407

**Pull request merged**